lib/services/datasets.js

```diff
@@ -0,0 +1,528 @@
+ *   if (!err) console.log('deleted!');
+ * });
+ */
+Datasets.prototype.deleteFeature = function(id, dataset, callback) {
+  invariant(typeof id === 'string', 'id must be a string');
+  invariant(typeof dataset === 'string', 'dataset must be a string');
+  invariant(typeof callback === 'function', 'callback must be a function');
+
+  var url = makeURL(this, constants.API_DATASET_FEATURE, {
+    user: this.user,
+    dataset: dataset,
+    id: id
+  });
+
+  request
+    .del(url)
+    .end(function(err, res) {
+      callback(err, res.body);
+    });
+};
+
+/**
+ * Perform a batch of inserts, updates, and deletes to a dataset in a single combined request.
+ * This request requires an access token with the datasets:write scope.
+ * There are a number of limits to consider when making this request:
+ *   - you can send a total of 100 changes (sum of puts + deletes) in a single request
+ *   - any single feature cannot be larger than 500 KB
+ *   - the dataset must not exceed 2000 total features
+ *   - the dataset must not exceed a total of 5 MB
+ *
+ * @param {Array<object>} put features to insert. Each feature must be a valid GeoJSON feature per http://geojson.org/geojson-spec.html#feature-objects
+ * @param {Array<string>} deletes ids of features to delete
+ * @param {string} dataset the id for an existing dataset
+ * @param {Function} callback called with (err, results)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * var inserts = [
+ *   {
+ *     "type": "Feature",
+ *     "properties": {
+ *       "name": "Null Island"
+ *     },
+ *     "geometry": {
+ *       "type": "Point",
+ *       "coordinates": [0, 0]
+ *     }
+ *   },
+ *   {
+ *     "type": "Feature",
+ *     "properties": {
+ *       "name": "Offshore from Null Island"
+ *     },
+ *     "geometry": {
+ *       "type": "Point",
+ *       "coordinates": [0.01, 0.01]
+ *     }
+ *   }
+ * ];
+ * var deletes =[
+ *   'feature-id-1',
+ *   'feature-id-2'
+ * ];
+ * mapboxClient.bulkFeatureUpdate(inserts, deletes, dataset, function(err, results) {
+ *  console.log(results);
+ * // {
+ * //   "put": [
+ * //     {
+ * //       "id": {feature-id},
+ * //       "type": "Feature",
+ * //       "properties": {
+ * //         "name": "Null Island"
+ * //       },
+ * //       "geometry": {
+ * //         "type": "Point",
+ * //         "coordinates": [0, 0]
+ * //       }
+ * //     },
+ * //     {
+ * //       "id": {feature-id},
+ * //       "type": "Feature",
+ * //       "properties": {
+ * //         "name": "Offshore from Null Island"
+ * //       },
+ * //       "geometry": {
+ * //         "type": "Point",
+ * //         "coordinates": [0.01, 0.01]
+ * //       }
+ * //     }
+ * //   ],
+ * //   "delete": [
+ * //     "feature-id-1",
+ * //     "feature-id-2"
+ * //   ]
+ * // }
+ * });
+ */
+Datasets.prototype.bulkFeatureUpdate = function(put, deletes, dataset, callback) {
```

Comment:
Yes, I'll work this in. I want to make sure that the names of the variables here match the properties of the response object returned from the API.